servers/rendering/renderer_rd/environment/fog.cpp

```diff
@@ -691,9 +691,9 @@ void Fog::volumetric_fog_update(const VolumetricFogSettings &p_settings, const P
 				// when the camera's near plane is inside the fog, we must always consider the entire screen
 				Vector3 near_plane_corner(frustum_near_size.x, frustum_near_size.y, z_near);
 				float expand = near_plane_corner.length();
-				if (cam_position.x > (corner_min.x - expand) && cam_position.x < (corner_max.x + expand) &&
-						cam_position.y > (corner_min.y - expand) && cam_position.y < (corner_max.y + expand) &&
-						cam_position.z > (corner_min.z - expand) && cam_position.z < (corner_max.z + expand)) {
+				if ((cam_position.x > (corner_min.x - expand) && cam_position.x < (corner_max.x + expand)) ||
```

Comment:
This change seems a bit aggressive to me. If any axis is within the range of the volume, then we calculate then we are applying the volume to the entire froxel buffer. This will have a huge performance impact as it will significantly reduce the situations where froxels are culled. 

Would it perhaps make more sense to change this to three conditions (one for each axis) and then only change the range for the corresponding axis?

Reply:
I'll admit that I made this change without really understanding exactly what the math is doing. Your suggestion seems good though, I'll give it a try and see what happens.

Reply:
Ok, My first pass at implementing this didn't work. The fog became visible from only certain angles when the camera was outside the Fog, though it seemed to behave correctly from inside. I've looked at the code some more, but I still don't really understand how this code is culling froxels so I'm a bit stymied. Is there a cull check methodology used for the AABB of normal objects? Since we know the position of each corner of each froxel could we use that same methodology on the froxels?

This is the code I tried, in case it's relevant.
```
// Tracking just the corners of the fog volume can result in missing some fog:
// when the camera's near plane is inside the fog, we must always consider the entire screen
Vector3 near_plane_corner(frustum_near_size.x, frustum_near_size.y, z_near);
float expand = near_plane_corner.length();
bool x_is_inside = (cam_position.x > (corner_min.x - expand) && cam_position.x < (corner_max.x + expand));
bool y_is_inside = (cam_position.y > (corner_min.y - expand) && cam_position.y < (corner_max.y + expand));
bool z_is_inside = (cam_position.z > (corner_min.z - expand) && cam_position.z < (corner_max.z + expand));
if ( x_is_inside || y_is_inside || z_is_inside) {
	if (x_is_inside) {
		froxel_min.x = 0;
		froxel_max.x = int32_t(fog->width);
	}
	if (y_is_inside) {
		froxel_min.y = 0;
		froxel_max.y = int32_t(fog->height);
	}
	if (z_is_inside) {
		froxel_min.z = 0;
		for (int j = 0; j < 8; j++) {
			froxel_max.z = MAX(froxel_max.z, froxels[j].z);
		}
	}
} else {
	// Camera is guaranteed to be outside the fog volume
	for (int j = 0; j < 8; j++) {
		froxel_min = froxel_min.min(froxels[j]);
		froxel_max = froxel_max.max(froxels[j]);
	}
}
```

Reply:
I made a minimal reproduction project for this bug: https://github.com/AshbyGeek/godot-fog-reproduction.

I also made a gif of both the old (godot 4.4 dev 4) behavior:
![godot-fog-recreation-old-behavior](https://github.com/user-attachments/assets/0690b4ff-f0d4-48c0-ada7-d8150616ca4c)

and the correct behavior based on commit 2e6ff1081f9a3869570ae09b58a47a06db185620 of this PR:
![godot-fog-recreation-correct-behavior](https://github.com/user-attachments/assets/a51a2463-9faf-4fad-81e0-6aaf14673cfb)